packages/shopping/src/controllers/user.controller.ts

```diff
@@ -260,6 +265,52 @@ export class UserController {
     // create a JSON Web Token based on the user profile
     const token = await this.jwtService.generateToken(userProfile);
 
+    // create a refreshtoken
+    const refreshtoken = await this.refreshtokenService.generateToken(
+      userProfile,
+    );
+
+    return {
+      token,
+      refreshtoken,
+    };
+  }
+
+  @post('/users/newtoken', {
+    security: OPERATION_SECURITY_SPEC,
+    responses: {
+      '200': {
+        description: 'Refreshing the token',
+        content: {
+          'application/json': {
+            schema: {
+              type: 'object',
+              properties: {
+                token: {
+                  type: 'string',
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  })
+  @authenticate('jwt')
+  // TODO(derdeka) find out why @authorize.allowAuthenticated() is not working
+  async refresh(
```

Comment:
When would the refresh function get called? I am trying to illustrate the flow of how a refresh token works: first a user login, then it gets the token and refresh token, then how does it decide when to call the `refresh` endpoint to extend the login time?

Reply:
> then how does it decide when to call the refresh endpoint to extend the login time?

In a `angular` application i would use a `serviceworker` or service with a timer to call this endpoint periodically in background.